File: RedSheet.DbEntities/Models/vTemplateModule.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTemplateModules", Schema = "dbo")]
	[Serializable]
	public partial class vTemplateModule  
	{

	public string AssigneeName { get; set; } 

	public string OwnerNote { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ProjectId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 

	public string ReviewerName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TemplateModuleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TemplateModuleName { get; set; } 

	public Nullable<int> UserId { get; set; } 

	public vTemplateModule()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-template-module.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTemplateModule {
    constructor(vTemplateModule?: vTemplateModule )  {
        let properties = [ "assigneeName", "ownerNote", "projectId", "projectModuleId", "reviewerName", "templateModuleId", "templateModuleName", "userId",];
        for (let property of properties)
            if (vTemplateModule && vTemplateModule[property])
                this[property] = vTemplateModule[property];
    }
 
	assigneeName : string =   undefined;
 
	ownerNote : string =   undefined;
 
	projectId : number =   0 ;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
	reviewerName : string =   undefined;
 
    @range(1,2147483647)
	templateModuleId : number =   undefined;
 
    @required()
    @maxLength(200)
	templateModuleName : string =   undefined;
 
	userId : number =   undefined;


}
